trin-core/src/utp/trin_helpers.rs

```diff
@@ -0,0 +1,107 @@
+// The code here isn't affiliated with stream and isn't part of the uTP implementation
+// this code is used in handling the uTP responses handled in events.rs.
+// just a clarification. It is just for handling the uTP stream higher up what stream releases.
+
+use ssz_derive::{Decode, Encode};
+
+// These Utp impl are related to sending messages over uTP not the implementation itself or stream
+pub struct UtpMessage {
+    pub length: u32,
+    pub payload: Vec<u8>,
+}
+
+impl UtpMessage {
+    pub fn new(payload: Vec<u8>) -> Self {
+        UtpMessage {
+            length: payload.len() as u32,
+            payload,
+        }
+    }
+
+    pub fn encode(&mut self) -> Vec<u8> {
+        let mut buf = vec![];
+        buf.extend_from_slice(&self.length.to_be_bytes());
+        buf.append(&mut self.payload);
+        buf
+    }
+
+    pub fn decode(bytes: &[u8]) -> Result<UtpMessage, String> {
+        if bytes.len() >= 4 {
+            let len = u32::from_be_bytes([bytes[0], bytes[1], bytes[2], bytes[3]]);
+            if len + 4 <= bytes.len() as u32 {
+                let mut payload_vec: Vec<u8> = vec![];
+                payload_vec.extend_from_slice(&bytes[4..(len as usize + 4)]);
```

Comment:
It's valid to just have ignored garbage at the end of the payload, past the stated length?

Reply:
Well, we will get the messages as it comes in, we might not end up using this since it isn't agreed upon how we will handle the stream yet, but the stream could be endless if they keep writing data to us. I like this approach because it means we could send multiple chunks of data.

I think we should have a length prefix, but we might just end up trying to decode whatever we have on close, it really isn't decided yet.